src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -773,225 +782,70 @@ private boolean invalidQuery(SearchPhaseExecutionException ex) {
 
     // For single entity detector
     class RCFActionListener implements ActionListener<RCFResultResponse> {
-        private List<RCFResultResponse> rcfResults;
         private String modelID;
         private AtomicReference<Exception> failure;
         private String rcfNodeID;
         private AnomalyDetector detector;
         private ActionListener<AnomalyResultResponse> listener;
-        private String thresholdModelID;
-        private DiscoveryNode thresholdNode;
         private List<FeatureData> featureInResponse;
-        private int nodeCount;
-        private final AtomicInteger responseCount;
         private final String adID;
-        private int numEnabledFeatures;
 
         RCFActionListener(
-            List<RCFResultResponse> rcfResults,
             String modelID,
             AtomicReference<Exception> failure,
             String rcfNodeID,
             AnomalyDetector detector,
             ActionListener<AnomalyResultResponse> listener,
-            String thresholdModelID,
-            DiscoveryNode thresholdNode,
             List<FeatureData> features,
-            int nodeCount,
-            AtomicInteger responseCount,
-            String adID,
-            int numEnabledFeatures
+            String adID
         ) {
-            this.rcfResults = rcfResults;
             this.modelID = modelID;
             this.rcfNodeID = rcfNodeID;
             this.detector = detector;
             this.listener = listener;
-            this.thresholdNode = thresholdNode;
-            this.thresholdModelID = thresholdModelID;
             this.featureInResponse = features;
             this.failure = failure;
-            this.nodeCount = nodeCount;
-            this.responseCount = responseCount;
             this.adID = adID;
-            this.numEnabledFeatures = numEnabledFeatures;
         }
 
         @Override
         public void onResponse(RCFResultResponse response) {
             try {
                 stateManager.resetBackpressureCounter(rcfNodeID, adID);
                 if (response != null) {
-                    rcfResults.add(response);
+                    listener
+                        .onResponse(
+                            new AnomalyResultResponse(
+                                response.getAnomalyGrade(),
+                                response.getConfidence(),
+                                response.getRCFScore(),
+                                featureInResponse,
+                                response.getTotalUpdates(),
+                                detector.getDetectorIntervalInMinutes()
+                            )
+                        );
                 } else {
                     LOG.warn(NULL_RESPONSE + " {} for {}", modelID, rcfNodeID);
+                    listener.onResponse(null);
```

Comment:
Can you confirm if the `listener` can handle `null` response?

Reply:
good catch.  Changed to "listener.onFailure(new InternalFailure(adID, NO_MODEL_ERR_MSG));"